app/src/services/cpk.ts

```diff
@@ -224,7 +245,20 @@ class CPKService {
       })
     }
     const txObject = await this.cpk.execTransactions(transactions, txOptions)
-    return this.waitForTransaction(txObject)
+    setTxState && setTxState(TransactionStep.transactionSubmitted)
+    setTxHash && setTxHash(txObject.hash)
+    const tx = await this.waitForTransaction(txObject)
+    setTxState && setTxState(TransactionStep.transactionConfirmed)
+    return tx
```

Comment:
@KadenZipfel Also probably not issue for this pr but i was thinking it could be useful maybe to create hook that takes transaction and returns the state and all relevant information needed? It could replace lot of code since we tend to use that pattern quite often or is it maybe too general to be made reusable?